### ISSUE TITLE
SearchBar fails to show keyboard on Android after coming back from the background

### ISSUE THREAD

**Issue by AndreHaueisen** [flutter/flutter#134965]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Click the SearchBar. They keyboard will come up
2. Send the app to the background
3. Come back from the background
4. Click the SearchBar. The keyboard will not show

Extra info:
I tested on two Android devices. On my Pixel 2 running Android 11 everything worked fine. The problem happened on my Pixel 6 running Android 13

### Expected results

When I come back from the background, I expect to see the keyboard again, or at least that it shows again if I click the SearchBar.

### Actual results

I can't call the keyboard back

### Code sample

<details><summary>Code sample</summary>

```dart
void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const Scaffold(
        body: Center(child: SearchBar()),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

https://drive.google.com/file/d/14rhlifS67w9rBBkJvMrqFUFonAoB8vOa/view?usp=sharing

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.4, on macOS 13.5.2 22G91 darwin-arm64
    (Rosetta), locale en-BR)
[!] Android toolchain - develop for Android devices (Android SDK version
    33.0.0-rc1)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] IntelliJ IDEA Community Edition (version 2023.2.2)
[✓] VS Code (version 1.80.1)
[✓] Connected device (4 available)
[✓] Network resources

```

</details>

**Comment by danagbemava-nc** [comment]
Hi @AndreHaueisen, I'm curious, do you only experience this when you use the `gesture-based` navigation in android?

**Comment by AndreHaueisen** [comment]
Hey, @danagbemava-nc. No, it also happens when the 3-button navigation is active

**Comment by huycozy** [comment]
@AndreHaueisen I checked this on my two devices (Realme 6 Android 11 and Pixel 4 Android 13) but can't reproduce the issue. 
Can you re-check this with a simple `TextField` widget? Please also confirm if the issue is reproducible on a newly created Flutter project or only on the current project.

**Comment by AndreHaueisen** [comment]
@huycozy The problem is specific to `SearchBar`. `TextField` works as expected.
I created a new project and tested it again. Same problem.

**Comment by AndreHaueisen** [comment]
Thanks for taking the time to test it, @danagbemava-nc 
This is definitely a weird one 🤔

**Comment by Renzo-Olivares** [comment]
I'm able to reproduce this on the `master` branch as well. I haven't been able to track down exactly why since it is a bit inconsistent but I think it has something to do with the fact that we `activate` the `TextField` by calling `requestFocus` on its `FocusNode` rather than interacting with the underlying `TextField` through `TextSelectionGestureDetector`. I'm not able to reproduce this issue on my PR that enables text selection in the search bar https://github.com/flutter/flutter/pull/137636 .

**Comment by Renzo-Olivares** [comment]
This was closed by #137636 which has landed on the `master` branch, if you are still experiencing this issue please feel free to re-open.
